syntax = "proto3";

package app.messages.events.v1;

import "app/messages/events/events.proto";
import "app/messages/events/v1/data.proto";

// SmsReceived is an event that is emitted when the system receives an SMS to send
message SmsReceived {
  option (messages.events.event_type) = "sms_received";
  messages.events.v1.Sms sms = 1;
}

// SmsSubmitted is an event that is emitted when submitting an SMS to a 3rd Party system to handle sending SMSes
message SmsSubmitted {
  option (messages.events.event_type) = "sms_submitted";
  messages.events.v1.Sms sms = 1;
}

// SmsSent is an event that is emitted when the system successfully sends out an SMS with a 3rd Part system
message SmsSent {
  option (messages.events.event_type) = "sms_sent";
  messages.events.v1.Sms sms = 1;
}
